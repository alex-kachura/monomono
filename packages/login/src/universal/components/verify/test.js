import React from 'react';
import { VerifyPage } from './';
import { shallow } from 'enzyme';

describe('[Component: VerifyPage]', () => {
  let component;

  const mockFields = [
    {
      name: 'mock-name',
      id: 'mock-id',
      label: 'mock label',
      isValid: true,
      value: '',
      type: 'input',
      constraints: [
        {
          type: 'mandatory',
          text: 'mock label',
          validator: true,
          isValid: true,
        },
        {
          type: 'regex',
          text: 'mock label',
          validationRegex: '^.{1,1}$',
          isValid: true,
        },
      ],
    }
  ];

  let mockProps = {
    fields: [],
    banner: {},
    stateToken: 'state-token',
    csrf: 'csrf-token',
    getLocalePhrase: (key) => key,
  };

  describe('account not locked', () => {
    describe('fields', () => {
      it('should render correctly', () => {
        mockProps = {
          ...mockProps,
          fields: mockFields,
        };
        component = shallow(<VerifyPage {...mockProps} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('no fields', () => {
      it('should render correctly', () => {
        component = shallow(<VerifyPage {...mockProps} />);

        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('account locked', () => {
    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
