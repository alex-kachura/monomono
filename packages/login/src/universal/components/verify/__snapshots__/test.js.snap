// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: VerifyPage] account locked should render correctly 1`] = `
<SideEffect(DocumentTitle)
  title="pages.verify.title"
>
  <div>
    <PageTitle
      margin={true}
      tagName="h1"
    >
      pages.verify.title
    </PageTitle>
    <BodyText
      emphasized={false}
      margin={true}
      tagName="p"
    >
      pages.verify.sub-title
    </BodyText>
    <Banner />
    <styled.div>
      <BodyText
        emphasized={false}
        margin={false}
        tagName="p"
      >
        pages.verify.digits-prompt
      </BodyText>
      <form
        autoComplete="off"
        method="POST"
        noValidate={true}
      >
        <styled.div
          key="mock-name"
        >
          <FormGroup
            disabled={false}
            error={false}
            errorMessage=""
            fieldset={false}
            helpLinkProps={Object {}}
            hideLabel={false}
            id="mock-id"
            labelProps={Object {}}
            labelText="mock label"
            required={true}
          >
            <Styled(Input)
              id="mock-id-input"
              name="mock-name"
              value=""
            />
          </FormGroup>
        </styled.div>
        <input
          defaultValue="state-token"
          id="state"
          name="state"
          type="hidden"
        />
        <input
          name="_csrf"
          type="hidden"
          value="csrf-token"
        />
        <Styled(WithProps(Button))
          type="submit"
          variant="primary"
        >
          pages.verify.buttons.submit
        </Styled(WithProps(Button))>
      </form>
    </styled.div>
  </div>
</SideEffect(DocumentTitle)>
`;

exports[`[Component: VerifyPage] account not locked fields should render correctly 1`] = `
<SideEffect(DocumentTitle)
  title="pages.verify.title"
>
  <div>
    <PageTitle
      margin={true}
      tagName="h1"
    >
      pages.verify.title
    </PageTitle>
    <BodyText
      emphasized={false}
      margin={true}
      tagName="p"
    >
      pages.verify.sub-title
    </BodyText>
    <Banner />
    <styled.div>
      <BodyText
        emphasized={false}
        margin={false}
        tagName="p"
      >
        pages.verify.digits-prompt
      </BodyText>
      <form
        autoComplete="off"
        method="POST"
        noValidate={true}
      >
        <styled.div
          key="mock-name"
        >
          <FormGroup
            disabled={false}
            error={false}
            errorMessage=""
            fieldset={false}
            helpLinkProps={Object {}}
            hideLabel={false}
            id="mock-id"
            labelProps={Object {}}
            labelText="mock label"
            required={true}
          >
            <Styled(Input)
              id="mock-id-input"
              name="mock-name"
              value=""
            />
          </FormGroup>
        </styled.div>
        <input
          defaultValue="state-token"
          id="state"
          name="state"
          type="hidden"
        />
        <input
          name="_csrf"
          type="hidden"
          value="csrf-token"
        />
        <Styled(WithProps(Button))
          type="submit"
          variant="primary"
        >
          pages.verify.buttons.submit
        </Styled(WithProps(Button))>
      </form>
    </styled.div>
  </div>
</SideEffect(DocumentTitle)>
`;

exports[`[Component: VerifyPage] account not locked no fields should render correctly 1`] = `
<SideEffect(DocumentTitle)
  title="pages.verify.title"
>
  <div>
    <PageTitle
      margin={true}
      tagName="h1"
    >
      pages.verify.title
    </PageTitle>
    <BodyText
      emphasized={false}
      margin={true}
      tagName="p"
    >
      pages.verify.sub-title
    </BodyText>
    <Banner />
    <styled.div>
      <BodyText
        emphasized={false}
        margin={false}
        tagName="p"
      >
        pages.verify.digits-prompt
      </BodyText>
      <form
        autoComplete="off"
        method="POST"
        noValidate={true}
      >
        <styled.div
          key="mock-name"
        >
          <FormGroup
            disabled={false}
            error={false}
            errorMessage=""
            fieldset={false}
            helpLinkProps={Object {}}
            hideLabel={false}
            id="mock-id"
            labelProps={Object {}}
            labelText="mock label"
            required={true}
          >
            <Styled(Input)
              id="mock-id-input"
              name="mock-name"
              value=""
            />
          </FormGroup>
        </styled.div>
        <input
          defaultValue="state-token"
          id="state"
          name="state"
          type="hidden"
        />
        <input
          name="_csrf"
          type="hidden"
          value="csrf-token"
        />
        <Styled(WithProps(Button))
          type="submit"
          variant="primary"
        >
          pages.verify.buttons.submit
        </Styled(WithProps(Button))>
      </form>
    </styled.div>
  </div>
</SideEffect(DocumentTitle)>
`;
